File: api/progress/definitions/progress.ts
export interface ProgressOptions {
    disable?: {
        movement?: boolean;
        vehicleMovement?: boolean;
        mouse?: boolean;
        combat?: boolean;
    };
    animation?: {
        dict: string;
        clip: string;
    };
    scenario?: string;
    onFinish?: () => void;
    onCancel?: () => void;
}

export interface ClientProgressProvider {
    startProgress(label: string, duration: number, options?: ProgressOptions): void;
    cancelProgress(): void;
}

File: api/progress/definitions/progress.lua
---@class ProgressOptions
---@field disable? { movement?: boolean, vehicleMovement?: boolean, mouse?: boolean, combat?: boolean }
---@field animation? { dict: string, clip: string }
---@field scenario? string
---@field onFinish? fun(): nil
---@field onCancel? fun(): nil

---@class ClientProgressProvider
---@field startProgress fun(label: string, duration: integer, options?: ProgressOptions): nil
---@field cancelProgress fun(): nil
